tests/api-diff/modified_variant_type_tests.t

```diff
@@ -0,0 +1,262 @@
+  diff module Modify_constructor_type:
+   type rank =
+    ...
+  - | Number of int
+  + | Number of float
+  
+  [1]
+
+Tests for modifying constructor's record argument
+
+  $ cat > shapes.mli << EOF
+  > type point = float * float
+  > type shape = 
+  >  | Circle of {center : point; raduis: int}
+  >  | Rectangle of {lower_left: point; upper_right: point}
+  > EOF
+
+We generate the .cmi file
+
+  $ ocamlc shapes.mli
+
+### Adding a field to a record type in a constructor argument:
+
+  $ cat > add_field.mli << EOF
+  > type point = float * float
+  > type shape = 
+  >  | Circle of {center : point; raduis: int; color:int}
+  >  | Rectangle of {lower_left: point; upper_right: point; color:int}
+  > EOF
+
+We generate the .cmi file
+
+  $ ocamlc add_field.mli 
+
+Run the api-watcher on the two cmi files
+
+  $ api-diff shapes.cmi add_field.cmi
+  diff module Add_field:
+   type shape =
+    ...
+    | Circle of {
+       ...
+  +    color : int;
+    }
+    | Rectangle of {
+       ...
+  +    color : int;
+    }
+  
+  [1]
+
+### Removing a field from a record type in a constructor argument:
+
+  $ cat > remove_field.mli << EOF
+  > type point = float * float
+  > type shape = 
+  >  | Circle of {center : point;}
+  >  | Rectangle of {lower_left: point; upper_right: point;}
+  > EOF
+
+We generate the .cmi file
+
+  $ ocamlc remove_field.mli 
+
+Run the api-watcher on the two cmi files
+
+  $ api-diff shapes.cmi remove_field.cmi
+  diff module Remove_field:
+   type shape =
+    ...
+    | Circle of {
+       ...
+  -    raduis : int;
+    }
+  
+  [1]
+
+### Modifying a field in a record type in a constructor argument:
+
+  $ cat > modify_field.mli << EOF
+  > type point = float * float
+  > type shape = 
+  >  | Circle of {center : point; raduis: float;}
+  >  | Rectangle of {lower_left: point; upper_right: point;}
+  > EOF
+
+We generate the .cmi file
+
+  $ ocamlc modify_field.mli 
+
+Run the api-watcher on the two cmi files
+
+  $ api-diff shapes.cmi modify_field.cmi
+  diff module Modify_field:
+   type shape =
+    ...
+    | Circle of {
+       ...
+  -    raduis : int;
+  +    raduis : float;
+    }
+  
+  [1]
+
+Tests for modifying constructor's tuple argument
+
+  $ cat > shapes2.mli << EOF
+  > type point = float * float
+  > type shape = 
+  >  | Circle of point * int
+  >  | Rectangle of point * point
+  > EOF
+
+We generate the .cmi file
+
+  $ ocamlc shapes2.mli
+
+### Adding an component to a tuple type in a constructor argument:
+
+  $ cat > add_component.mli << EOF
+  > type point = float * float
+  > type shape = 
+  >  | Circle of point * int * int
+  >  | Rectangle of point * point * int
+  > EOF
+
+We generate the .cmi file
+
+  $ ocamlc add_component.mli 
+
+Run the api-watcher on the two cmi files
+
+  $ api-diff shapes2.cmi add_component.cmi
+  diff module Add_component:
+   type shape =
+    ...
+    | Circle of
+       point *
+       int *
+  +    int
+    | Rectangle of
```

Comment:
I still believe this should be displayed on a single line!

Reply:
Shouldn't the inling of text diffs be done in a separate PR ?

Reply:
I'm not suggesting any highlighting here, simply that the line based diff is showed as:
```
-  | Circle of point * int
+  | Circle of point * int * int
```
as that is simply more readable.